dns-server/src/storage.rs

```diff
@@ -685,16 +774,38 @@ pub(crate) enum QueryError {
     #[error("server is not authoritative for name: {0:?}")]
     NoZone(String),
 
-    #[error("no records found for name: {0:?}")]
-    NoName(String),
```

Comment:
Is this an externally-visible behavior change?

Reply:
no, this case turns into the `if name_records.is_empty() { return Err(... NxDomain ...) }` check on the caller side, which still results in the NxDomain we'd expect. the one exception being if the query is for the zone apex, which is an intentional change :grin: